tools/walletextension/wallet_extension.go

```diff
@@ -153,105 +131,111 @@ func (we *WalletExtension) handleHTTPEthJSON(resp http.ResponseWriter, req *http
 		return
 	}
 
-	// We unmarshall the JSON request.
-	var reqJSONMap map[string]interface{}
+	// We unmarshall the JSON request
+	var reqJSONMap map[string]json.RawMessage
 	err = json.Unmarshal(body, &reqJSONMap)
 	if err != nil {
-		logAndSendErr(resp, fmt.Sprintf("could not unmarshall JSON-RPC request body to JSON: %s. "+
+		logAndSendErr(resp, fmt.Sprintf("could not unmarshal JSON-RPC request body to JSON: %s. "+
 			"If you're trying to generate a viewing key, visit %s", err, pathViewingKeys))
 		return
 	}
-	method := reqJSONMap[reqJSONKeyMethod]
-	fmt.Printf("Received %s request from wallet: %s\n", method, body)
 
-	reqJSONMap, err = we.ensureCallsHaveFromField(method, reqJSONMap)
+	var reqID string
+	err = json.Unmarshal(reqJSONMap[reqJSONKeyID], &reqID)
 	if err != nil {
-		logAndSendErr(resp, err.Error())
+		logAndSendErr(resp, fmt.Sprintf("could not unmarshal id from JSON-RPC request body: %s", err))
 		return
 	}
-
-	// We encrypt the request's params with the enclave's public key if it's a sensitive request.
-	maybeEncryptedBody, err := we.encryptParamsIfNeeded(body, method, reqJSONMap)
+	var method string
+	err = json.Unmarshal(reqJSONMap[reqJSONKeyMethod], &method)
 	if err != nil {
-		logAndSendErr(resp, fmt.Sprintf("could not encrypt request parameters: %s", err))
+		logAndSendErr(resp, fmt.Sprintf("could not unmarshal method string from JSON-RPC request body: %s", err))
 		return
 	}
+	fmt.Printf("Received %s request from wallet: %s\n", method, body)
 
-	// We forward the request on to the Obscuro node.
-	nodeResp, err := forwardMsgOverWebsocket(websocketProtocol+we.hostAddr, maybeEncryptedBody)
+	params, err := we.extractAndPrepareParams(method, reqJSONMap)
 	if err != nil {
-		logAndSendErr(resp, fmt.Sprintf("received error response when forwarding request to node at %s: %s", we.hostAddr, err))
+		logAndSendErr(resp, err.Error())
 		return
 	}
 
-	// We unmarshall the JSON response.
-	var respJSONMap map[string]interface{}
-	err = json.Unmarshal(nodeResp, &respJSONMap)
+	respStr := ""
+	err = we.hostClient.Call(&respStr, method, params...)
 	if err != nil {
-		logAndSendErr(resp, fmt.Sprintf("could not unmarshall enclave response to JSON: %s", err))
+		logAndSendErr(resp, fmt.Sprintf("rpc request failed: %s", err))
 		return
 	}
 
-	// We report any errors from the request.
-	if respJSONMap[respJSONKeyErr] != nil {
-		logAndSendErr(resp, respJSONMap[respJSONKeyErr].(map[string]interface{})[respJSONKeyMsg].(string))
-		return
-	}
+	respMap := make(map[string]interface{})
+	respMap[resJSONKeyID] = reqID
```

Comment:
@BedrockSquirrel What's this addition about? The response should be coming from the host properly formatted already (e.g. with the ID and the JSON version.

Reply:
Oh, I see. We're now making an explicit JSON call, rather than passing something back and forth.